be/src/exec/write_combined_txn_log.cpp

```diff
@@ -14,14 +14,68 @@
 
 #include "exec/write_combined_txn_log.h"
 
+#include <future>
+#include <vector>
+
 #include "runtime/exec_env.h"
 #include "storage/lake/tablet_manager.h"
 
 namespace starrocks {
 
+class TxnLogTask : public Runnable {
+public:
+    TxnLogTask(const CombinedTxnLogPB* logs, lake::TabletManager* tablet_mgr, std::promise<Status> promise)
+            : _logs(logs), _tablet_mgr(tablet_mgr), _promise(std::move(promise)) {}
+
+    void run() override {
+        try {
+            Status status = _tablet_mgr->put_combined_txn_log(*_logs);
+            if (!status.ok()) {
+                throw std::runtime_error("Log write failed");
+            }
+            _promise.set_value(Status::OK());
+        } catch (const std::exception& e) {
+            _promise.set_value(Status::IOError(e.what()));
+        }
+    }
+
+private:
+    const CombinedTxnLogPB* _logs;
+    lake::TabletManager* _tablet_mgr;
+    std::promise<Status> _promise;
+};
+
 Status write_combined_txn_log(const CombinedTxnLogPB& logs) {
     auto tablet_mgr = ExecEnv::GetInstance()->lake_tablet_manager();
     return tablet_mgr->put_combined_txn_log(logs);
 }
 
+Status write_combined_txn_log(const std::map<int64_t, CombinedTxnLogPB>& txn_log_map) {
+    std::vector<std::future<Status>> futures;
+    std::vector<std::shared_ptr<Runnable>> tasks;
+
+    for (const auto& [partition_id, logs] : txn_log_map) {
+        (void)partition_id;
+        std::promise<Status> promise;
+        futures.push_back(promise.get_future());
+        std::shared_ptr<Runnable> r(
+                std::make_shared<TxnLogTask>(&logs, ExecEnv::GetInstance()->lake_tablet_manager(), std::move(promise)));
+        tasks.emplace_back(std::move(r));
+    }
+
+    for (auto& task : tasks) {
```

Comment:
task submit to thread pool may not necessary get executed. The `future.get()` may never be return.

Reply:
Add a timeout to ensure it doesn't block indefinitely.

Reply:
it may not fair to wait_for for each promise resulting the accumulative timeout will be way long than `parallel_write_combinded_txn_log_timeout_sec`.  A possible solution could be leveraging the CountDownLatch(n), each task will latch.countdown(), here will just wait for latch.countdown to zero.

Also check about `AutoCleanRunable` in util/threadpool.h to ensure every task will be countdown no matter it get executed or get abandoned.

Reply:
> it may not fair to wait_for for each promise resulting the accumulative timeout will be way long than `parallel_write_combinded_txn_log_timeout_sec`. A possible solution could be leveraging the CountDownLatch(n), each task will latch.countdown(), here will just wait for latch.countdown to zero.
> 
> Also check about `AutoCleanRunable` in util/threadpool.h to ensure every task will be countdown no matter it get executed or get abandoned.

done.